### DISCUSSION TITLE
App Development: Framework Usage and Team Divisions

### DISCUSSION THREAD

**Discussion by YanehRuiz** [uprm-inso4116-2024-2025-s1/semester-project-SafeRUM#3] [Thoughts]
Good evening everyone,

As we move forward with the development of our mobile app, we need to make some key decisions and align our approach. Firstly, I would like to hear everyone's opinion on the framework preferred to work with. The proposed framework is react, however, let us know if there's another framework you're more comfortable working with.

In addition, let's discuss the division of teams based on features to ensure clear responsibilities and effective collaboration. The proposed divisions are as follows:

-     User Interface Team: Focused on Sign In/Log In functionalities.
-     Admin Interface Team: Dedicated to developing the UPRM guards Communication features.

Within these teams, there are also specific subgroups that need to be established:

-      Location Management: This subgroup will be part of the User Interface group and will handle either the locations of reports and boundaries with coordinates to identify buildings or user input related to locations.
-     Trust System: This subgroup is part of the Admin Interface group. Responsible for managing the "rumors" (unconfirmed alerts), sighted reports, and tracking trust levels or user input.

If you're interested in becoming a Team Leader, please leave a comment below with which team you're interested in. Also, remember to vote for which framework to use, React or another (Comment which one!) and we'll poll them.

**Comment by ghost** [comment]
I think that for this project we might require some kind of server in order to communicate with clients.

So in my mind, someone will report an event they saw along with the details, the client would send that event to a server, then other clients would constantly poll the server for new events, and fetch the events from it, notifying the user of them.

We would also need to implement a database in the server to save all the events, and a database in the client.

In essence, the server would just act as a simple api for the database that all clients can refer to for all the events that have been reported.

If we decide on creating a server like this, then I think we could write it in Go.

**Comment by evnd04** [comment]
We have now decided to instead of having subgroups these will be their own groups, we will detail the features for each group so you have an objective for the milestone before it starts on Friday!
